azurelinuxagent/common/future.py

```diff
@@ -128,6 +128,17 @@ def get_openwrt_platform():
                     result[0] = "openwrt"
     return result
 
+def is_file_not_found_error(exception):
+    # Python 2 uses IOError(errno=2)
+    if sys.version_info[0] == 2:
+        return isinstance(exception, IOError) and exception.errno == 2
+    
+    # Python 3 has its own type for this error.
+    elif sys.version_info[0] == 3:
+        return isinstance(exception, FileNotFoundError)
+    
+    else:
```

Comment:
not sure if we should have this condition here. the module already protects against unknow versions. having a duplicate may introduce issues if one check is updated but not the other